### ISSUE TITLE
Image Tab not showing any images

### ISSUE THREAD

**Issue by jburleson** [specify/webportal-installer#7]
I have the portal up and running with the exception of the Images tab. From what I can tell using the Firefox tools, here is one entry that is returning a '400 Bad Response'.

`http://example.com:8081/portal/vascularplants/select?indent=on&version=2.2&fq=&rows=10000&fl=spid,cn,fn,img%2Cscore&qt=&wt=json&explainOther=&hl.fl=&q=_query_:%22*%22+AND+_query_:%22img:[\%22\%22%20TO%20^]%22&_dc=1488299702848&page=1&start=0&limit=10000&json.wrf=Ext.data.JsonP.callback2`

The attachment server is running and it is serving up images as I can double click on one of the records and view the image but for some reason I am not able to get the image tab to display anything.

I am also seeing this message in the catalina.out log.
`Feb 28, 2017 10:24:38 AM org.apache.coyote.http11.AbstractHttp11Processor process
INFO: Error parsing HTTP request header
 Note: further occurrences of HTTP header parsing errors will be logged at DEBUG level.`

Any thoughts?

**Comment by benanhalt** [comment]
Hmm. I just tried that request against our portal server and didn't get the 400 error.

Some Googling leads me to suspect the URL may be making the HTTP headers too long for your TomCat server as a possibility. Does increasing `maxHttpHeaderSize` in the Connector node in `server.xml`make any difference?

I guess you could also find the full HTTP request header in the Firefox dev tools network tab and see if anything looks suspicious.

**Comment by jburleson** [comment]
Yeah, I saw that as well on Google. I did try increasing the `maxHttpHeaderSize` to 64K and then to 1024K just to be sure, but it still returned a `400 Bad Request`.

I have also already taken a look at the raw headers and did not see anything unusual about them. I even compared them to other web portals and they looked the same.

I will keep looking. Thanks for the suggestion.

**Comment by jburleson** [comment]
I do not know much about solr so this might not be the right way to go but i was able to fix this by changing the `getImageRequirementFilter` in PortalApp/app/store/MainSolrStore.js.

I changed it from `return 'img:[\\"\\" TO ^]';'` to `return 'img:[* TO *]';`

**Comment by benanhalt** [comment]
What version of Solr are you using? I wonder if the query syntax changed since it is accepted by our solr instance which is 4.7.2, I think.

**Comment by benanhalt** [comment]
@timo11 is this something you have seen before?

**Comment by jburleson** [comment]
It is 4.7.2.

**Comment by timo11** [comment]
I've definitely seen syntax change between versions. Maybe in this case 
there's a configuration setting causing it to behave differently?

**Comment by jburleson** [comment]
I followed the Auto Update Instructions. Outside of that I am running the following:

OS: Ubuntu 14.04 x86_64
Tomcat: 7.0.52
Java: Oracle Java 8 -- I also tried th edefault-jdk package (openjpdk-7) but it had the same results.

**Comment by benanhalt** [comment]
@jburleson, were you able to resolve this issue? We recently experienced a similar problem with `400 Bad Request` responses that turned out to be due to a Tomcat bug. I resolved it by updating to Ubuntu 16.06.

**Comment by rafaelmr2** [comment]
Hi,
I'm using Ubuntu 16.04.3 updated and I'm having the same issue. Images are not showing anymore.

Thanks,

**Comment by timo11** [comment]
tomcat 7.0.68 is known to work with the Specify Web Portal

**Comment by benanhalt** [comment]
It seems like the problem is with parsing the image present query term. I tried changing it to `img:[* TO *]` here:
https://github.com/specify/webportal-installer/blob/5b0f2fdf9036e536355a5e551a31731070777217/PortalApp/app/store/MainSolrStore.js#L145

It seems to work OK. @timo11 do you know any reason why this syntax wouldn't work?

I found it on this page https://lucene.apache.org/solr/guide/6_6/the-standard-query-parser.html

**Comment by timo11** [comment]
No reason I know of the img change won't work.
There is also a problem parsing the query that retrieves specimens at a marker on the map view. (I think I found a partial fix for that a few months ago.)

**Comment by timo11** [comment]
img:[* TO *] didn't work for me with tomcat 8.5 or tomcat 7 (the highest version I think), but img:* does.
That change is checked in. Still haven't figured out how to make queries for map markers with negative geocoords more parseable.
